unit-tests/py/rspy/acroname.py

```diff
@@ -41,347 +41,344 @@ def usage():
-    """
-    Return all Acroname module specs in a list. Raise NoneFoundError if one is not found!
-    """
+    hub = None
 
-    log.d( 'discovering Acroname modules ...' )
-    # see https://acroname.com/reference/_modules/brainstem/module.html#Module.discoverAndConnect
-    try:
-        log.debug_indent()
-        for i in range(retries + 1):
-            specs = brainstem.discover.findAllModules( brainstem.link.Spec.USB )
-            if not specs:
-                time.sleep(1)
-            else:
-                for spec in specs:
-                    log.d( '...', spec )
-                break
-        if not specs:
+    def __init__(self):
+        super().__init__()
+        acroname = self.discover()
+        if acroname is None:
             raise NoneFoundError()
-    finally:
-        log.debug_unindent()
-
-    return specs
-
-
-def connect( reset = False, req_spec = None ):
-    """
-    Connect to the hub. Raises RuntimeError on failure
-    :param reset: When true, the acroname will be reset as part of the connection process
-    :param req_spec: Required spec to connect to.
-    """
-
-    global hub
-    if not hub:
-        hub = brainstem.stem.USBHub3p()
-
-    if req_spec:
-        specs = [req_spec]
-    else:
-        specs = discover()
-
-    spec = specs[0]
-    result = hub.connectFromSpec( spec )
-    if result != brainstem.result.Result.NO_ERROR:
-        raise RuntimeError( "failed to connect to Acroname (result={})".format( result ))
-    elif len(specs) > 1:
-        log.d( 'connected to', spec )
-
-    if reset:
-        log.d("resetting Acroname...")
-        result = hub.system.reset()
-        # According to brainstem support:
-        # Result error is expected, so we do not check it
-        # * there is also a brainstem internal console print on Linux "error release -4"
-        # Disconnection is needed after a reset command
-        hub.disconnect()
-
-        result = None
-        for i in range(10):
-            result = hub.connectFromSpec(spec)
-            if result != brainstem.result.Result.NO_ERROR:
-                time.sleep(1)
-            else:
-                log.d('reconnected')
-                return
-        raise RuntimeError("failed to reconnect to Acroname (result={})".format(result))
-
-
-
-
-def find_all_hubs():
-    """
-    Yields all hub port numbers
-    """
-    from rspy import lsusb
-    #
-    # 24ff:8013 =
-    #   iManufacturer           Acroname Inc.
-    #   iProduct                USBHub3p-3[A]
-    hubs = set( lsusb.devices_by_vendor( '24ff' ))
-    ports = set()
-    # go thru the tree and find only the top-level ones (which we should encounter first)
-    for dev,dev_port in lsusb.tree():
-        if dev not in hubs:
-            continue
-        for port in ports:
-            # ignore everything inside existing hubs - we only want the top-level
-            if dev_port.startswith( port + '.' ):
-                break
+
+    def discover(self, retries = 0):
```

Comment:
This looks like a static function, not an instance function.
I.e., it does not change the state of the instance.
Let's take it out of the class.